### PR TITLE
Adding get source table to the module

### PR DESCRIPTION
**Pull request by schafer14**

### CODE REVIEW

File: examples_test.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package ntrip
+
+import (
+	"context"
+	"fmt"
+)
+
+func ExampleGetSourcetable() {
```
Comment (umeat) [inline examples_test.go:8]: Nice. I've been meaning to set some of these up. Cheers.

File: sourcetable.go [hunk 2]
```diff
@@ -128,6 +131,44 @@ func (m StreamEntry) String() string {
 		m.Authentication, fee, m.Bitrate, m.Misc)
 }
 
+// GetSourcetable fetches a source table from a specific caster.
+//
+// The funciton returns a list of errors which can be treated as warnings.
+// These warnings indicate that the caster is returning an improper rtcm3 format.
```
Comment (umeat) [inline sourcetable.go:137]: Sourcetable is not RTCM.

File: sourcetable.go [hunk 3]
```diff
@@ -128,6 +131,44 @@ func (m StreamEntry) String() string {
...
+		return Sourcetable{}, warnings, err
+	}
+
+	if res.StatusCode != 200 {
+		return Sourcetable{}, warnings, fmt.Errorf("received a non 200 status code")
+	}
+
+	// Swollowing the errors here is okay because the errors are more like warnings.
+	// All rows that could be parsed will be present in the source table.
+	table, warnings := ParseSourcetable(string(body[:]))
```
Comment (umeat) [inline sourcetable.go:168]: It's a little weird that if the response body is not at all a sourcetable (just some other content entirely) the error will be nil (with a bunch of parse errors).

It's definitely more convenient for us to do it this way, so we're not stopped from parsing the information we care about if the first line is a malformed CAS entry or whatever.

Comment (schafer14) [reply sourcetable.go:168]: I agree kind of. I considered three signatures for the ParseSourcetable (not returning any errors, returning a list of warnings, and returning a list of warnings and an error). I think the last option would handle the situation where you got back random data better, but I thought it added complexity for a use case that would almost never occur.

Happy to change it if you think that is better.

Comment (umeat) [reply sourcetable.go:168]: No, that's all good. I agree it's probably not necessary. Happy to go with this approach.

The only other approach that I think would be reasonable would be much more heavy-handedly returning an error as soon as you get a parsing error. Where the rationale is that you Get a valid Sourcetable, or you error - and if the Sourcetable contains any invalid lines then it is invalid.

Again, this is good.

**Review by umeat — APPROVED** [review]

File: sourcetable.go (hunk 3 above)
Comment (schafer14) [reply sourcetable.go:168]: Awesome. I'm going to move back to gnss-stream for a bit.

**Review by umeat — APPROVED** [review]

**Pull request merged**